sync-core/src/main/java/com/cloudant/sync/datastore/PreparedAttachment.java

```diff
@@ -43,20 +47,47 @@ public PreparedAttachment(Attachment attachment,
                               String attachmentsDir) throws AttachmentException {
         this.attachment = attachment;
         this.tempFile = new File(attachmentsDir, "temp" + UUID.randomUUID());
-        FileInputStream tempFileIS = null;
+        InputStream attachmentInStream = null;
+        OutputStream tempFileOutStream = null;
+        MessageDigest sha1 = null;
         try {
-            FileUtils.copyInputStreamToFile(attachment.getInputStream(), tempFile);
-            this.sha1 = Misc.getSha1((tempFileIS = new FileInputStream(tempFile)));
+            attachmentInStream = attachment.getInputStream();
+
+            //Use FileUtils to create folder structure and file if necessary for output stream
+            tempFileOutStream = FileUtils.openOutputStream(this.tempFile);
+
+            sha1 = MessageDigest.getInstance("SHA-1");
+            int bufSiz = 1024;
+            byte buf[] = new byte[bufSiz];
+            int bytesRead;
+            int totalRead = 0;
+            while ((bytesRead = attachmentInStream.read(buf)) != -1) {
+                sha1.update(buf, 0, bytesRead);
+                tempFileOutStream.write(buf, 0, bytesRead);
+                totalRead += bytesRead;
+            }
+
+            //Set attachment length from bytes read in input stream
+            this.length = totalRead;
```

Comment:
Probably set this down by where the sha1 is set for the object, for consistency.